src/main/java/uk/gov/hmcts/ccd/v2/internal/controller/UICaseSearchController.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.ccd.v2.internal.controller;
+
+import io.swagger.annotations.*;
+import lombok.extern.slf4j.*;
+import org.springframework.beans.factory.annotation.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.hmcts.ccd.auditlog.AuditOperationType;
+import uk.gov.hmcts.ccd.auditlog.LogAudit;
+import uk.gov.hmcts.ccd.domain.model.search.*;
+import uk.gov.hmcts.ccd.domain.model.search.elasticsearch.SearchResultViewItem;
+import uk.gov.hmcts.ccd.domain.model.search.elasticsearch.UICaseSearchResult;
+import uk.gov.hmcts.ccd.domain.service.search.elasticsearch.*;
+import uk.gov.hmcts.ccd.domain.service.search.elasticsearch.security.*;
+import uk.gov.hmcts.ccd.v2.*;
+import uk.gov.hmcts.ccd.v2.internal.resource.*;
+
+import java.time.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.auditlog.aop.AuditContext.CASE_ID_SEPARATOR;
+import static uk.gov.hmcts.ccd.auditlog.aop.AuditContext.MAX_CASE_IDS_LIST;
+
+@RestController
+@RequestMapping(path = "/internal/searchCases")
+@Slf4j
+public class UICaseSearchController {
+    private static final String ERROR_CASE_ID_INVALID = "Case ID is not valid";
+
+    private final CaseSearchOperation caseSearchOperation;
+    private final ElasticsearchQueryHelper elasticsearchQueryHelper;
+
+    @Autowired
+    public UICaseSearchController(
+        @Qualifier(AuthorisedCaseSearchOperation.QUALIFIER) CaseSearchOperation caseSearchOperation,
+        ElasticsearchQueryHelper elasticsearchQueryHelper) {
+        this.caseSearchOperation = caseSearchOperation;
+        this.elasticsearchQueryHelper = elasticsearchQueryHelper;
+    }
+
+    @PostMapping(
+        path = "",
+        headers = {
+            V2.EXPERIMENTAL_HEADER
+        },
+        produces = {
+            V2.MediaType.UI_CASE_SEARCH
```

Comment:
I know other v2 apis have versioned custom vendor types, but I'd like to follow HMCTS guidelines from now on which says API versioning should be avoided. We can then use a generic media type `application/json`. See:
https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1392411337#RDM-8325:InternalSearchAPI-3.1-Vendor-SpecificMediaTypes